portfolio/src/main/java/com/google/sps/servlets/GetExpiryDateServlet.java

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Refrigerator;
+import com.google.sps.data.User;
+import com.google.sps.data.Item;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing tasks. */
+@WebServlet("/get-expiry-date")
+public class ListTasksServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Refrigerator");
+    String idNumber = request.getParameter("id");
+    String item = request.getParameter("item");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    String expiry_date = "";
+    for (Entity entity : results.asIterable()) {
+      String id = (String) entity.getProperty("id");
+      if (id.equals(idNumber)) {
+          ArrayList<Item> items = (String) entity.getProperty("items");
+           for (int i = 0; i < items.size(); i++) {
+               if (items.get(i).equals(item)){
+                expiry_date = items.get(i).getExpiryDate();
+               }
+           }
```

Comment:
Again you don't need to loop through all the items. Get the item from the arraylist using indexOf() and get() and then return.